ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -718,33 +718,40 @@ func (r *RayServiceReconciler) updateServeDeployment(ctx context.Context, raySer
 	}
 }
 
-// getAndCheckServeStatus gets Serve applications' and deployments' statuses,
-// updates health timestamps, and checks if the RayCluster is overall healthy.
-// It takes as one of its inputs `serveConfigType`, which is used to decide whether
-// to query the single-application Serve REST API or the multi-application Serve REST API
-// It's return values should be interpreted as
-// (Serve app healthy?, Serve app ready?, error if any)
+// `getAndCheckServeStatus` gets Serve applications' and deployments' statuses, updates health timestamps,
+// and checks if the RayCluster is overall healthy. It takes as one of its inputs `serveConfigType`, which
+// is used to decide whether to query the single-application Serve REST API or the multi-application Serve
+// REST API. It's return values should be interpreted as:
+//
+// (Serve app healthy?, Serve app ready?, error if failed to get Serve statuses)
 func (r *RayServiceReconciler) getAndCheckServeStatus(ctx context.Context, dashboardClient utils.RayDashboardClientInterface, rayServiceServeStatus *rayv1alpha1.RayServiceStatus, serveConfigType utils.RayServeConfigType, unhealthySecondThreshold *int32) (bool, bool, error) {
-	// If the unhealthySecondThreshold value is non-nil, then we will use that value.
-	// Otherwise, we will use the value ServiceUnhealthySecondThreshold which can be set in a test
-	// This is used for testing purposes.
+	// If the `unhealthySecondThreshold` value is non-nil, then we will use that value. Otherwise, we will use the value ServiceUnhealthySecondThreshold
+	// which can be set in a test. This is used for testing purposes.
 	serviceUnhealthySecondThreshold := ServiceUnhealthySecondThreshold
 	if unhealthySecondThreshold != nil {
 		serviceUnhealthySecondThreshold = float64(*unhealthySecondThreshold)
 	}
 
+	// TODO (kevin85421): Separate the logic for retrieving Serve application statuses from checking Serve application statuses into two separate functions.
+	// Currently, the handling logic for `isHealthy` and `isReady` between these two behaviors is inconsistent. This can cause potential issues in the future.
```

Comment:
Agree the logic is confusing, should we file a tracking issue on github?

Reply:
Track the progress https://github.com/ray-project/kuberay/issues/1234.